src/FRP/Event/Class.purs

```diff
@@ -67,3 +69,21 @@ mapAccum f xs acc = filterMap snd
 -- | the second event.
 sampleOn_ :: forall event a b. IsEvent event => event a -> event b -> event a
 sampleOn_ a b = sampleOn a (b $> id)
+
+-- | Sample the events that are fired while a boolean event is true.
+gate :: forall a event. IsEvent event => event Boolean -> event a -> event a
+gate = gateBy const
+
+-- | Generalised form of `gateBy`, allowing for any predicate between
+-- | the two events. When true, the second event is sampled.
+gateBy
+  :: forall a b event.
+     IsEvent event
+  => (a -> b -> Boolean)
+  -> event a
+  -> event b
+  -> event b
+gateBy f s
+   = filtered
+ <<< sampleOn s
```

Comment:
One issue here is that `b`s will be dropped until we see the first `a`. I think that needs fixing or documenting.

Reply:
Yes, good point; which would you prefer? I'll take a look at the semantics, too!

Reply:
I think I would prefer to not drop things initially, but it could even be indicated using a function argument.